src/button/ReactionToggleButton.tsx

```diff
@@ -0,0 +1,324 @@
+} from "../reactions";
+import { Modal } from "../Modal";
+
+interface InnerButtonProps extends ComponentPropsWithoutRef<"button"> {
+  raised: boolean;
+  open: boolean;
+}
+
+const InnerButton: FC<InnerButtonProps> = ({ raised, open, ...props }) => {
+  const { t } = useTranslation();
+
+  return (
+    <Tooltip label={t("action.raise_hand_or_send_reaction")}>
+      <CpdButton
+        className={classNames(raised && styles.raisedButton)}
+        aria-expanded={open}
+        kind={raised || open ? "primary" : "secondary"}
+        iconOnly
+        Icon={raised ? RaisedHandSolidIcon : ReactionIcon}
+        {...props}
+      />
+    </Tooltip>
+  );
+};
+
+export function ReactionPopupMenu({
+  sendReaction,
+  toggleRaisedHand,
+  isHandRaised,
+  canReact,
+}: {
+  sendReaction: (reaction: ReactionOption) => void;
+  toggleRaisedHand: () => void;
+  isHandRaised: boolean;
+  canReact: boolean;
+}): ReactNode {
+  const { t } = useTranslation();
+  const [searchText, setSearchText] = useState("");
+  const [isSearching, setIsSearching] = useState(false);
+  const onSearch = useCallback<ChangeEventHandler<HTMLInputElement>>((ev) => {
+    ev.preventDefault();
+    setSearchText(ev.target.value.trim().toLocaleLowerCase());
+  }, []);
+
+  const filteredReactionSet = useMemo(
+    () =>
+      ReactionSet.filter(
+        (reaction) =>
+          !isSearching ||
+          (!!searchText &&
+            (reaction.name.startsWith(searchText) ||
+              reaction.alias?.some((a) => a.startsWith(searchText)))),
+      ).slice(0, 6),
+    [searchText, isSearching],
+  );
+
+  const onSearchKeyDown = useCallback<KeyboardEventHandler<never>>(
+    (ev) => {
+      if (ev.key === "Enter") {
+        ev.preventDefault();
+        if (!canReact) {
+          return;
+        }
+        if (filteredReactionSet.length !== 1) {
+          return;
+        }
+        sendReaction(filteredReactionSet[0]);
+        setIsSearching(false);
+      } else if (ev.key === "Escape") {
+        ev.preventDefault();
+        setIsSearching(false);
+      }
+    },
+    [sendReaction, filteredReactionSet, canReact, setIsSearching],
+  );
+
+  return (
+    <div className={styles.reactionPopupMenu}>
+      <section className={styles.handRaiseSection}>
+        <Tooltip label={t("common.raise_hand")}>
+          <CpdButton
+            kind={isHandRaised ? "primary" : "secondary"}
+            aria-pressed={isHandRaised}
+            aria-label="Toggle hand raised"
+            onClick={() => toggleRaisedHand()}
+            iconOnly
+            Icon={RaisedHandSolidIcon}
+          />
+        </Tooltip>
+      </section>
+      <div className={styles.verticalSeperator} />
+      <section>
+        {isSearching ? (
+          <>
+            <Form.Root className={styles.searchForm}>
+              <Search
+                required
+                value={searchText}
+                name="reactionSearch"
+                placeholder="Search reactions…"
```

Comment:
Untranslatable

Reply:
https://github.com/element-hq/element-call/issues/2735 suggested so this doesn't happen by accident.